packages/cache/src/index.ts

```diff
@@ -3,27 +3,42 @@ import { AzureBlobCacheStorage } from "./AzureBlobCacheStorage";
 import { LocalCacheStorage } from "./LocalCacheStorage";
 import { NpmCacheStorage } from "./NpmCacheStorage";
 
-import { CacheStorageConfig } from "backfill-config";
+import { CacheStorageConfig, BackfillModes } from "backfill-config";
 
 export { ICacheStorage } from "./CacheStorage";
 
+function applyMode(cacheStorage: ICacheStorage, mode: BackfillModes) {
```

Comment:
It feels a bit weird to go change the implementation of a class outside of this class. Do we really need to handle these cases at the cache level?

Reply:
Do you have a better idea for where to solve this? In my view, one alternative I was considering was to conditionally invoke fetch/put depending on the mode, but that would convolute the main backfill function (IMO).

Reply:
This is absolutely not important by the way, just a style/preference comment. I would put this logic in the main backfill function and refactoring this one if need be.

Reply:
Ok. Then I'll punt on this, for now, discuss it with you in more detail later and then refactor.